### PR TITLE
For reference docs - Fix for github issue 1290 and remove hardcoded EN lang links

### PR DESCRIPTION
**Pull request by wesmc7777**
# Checklist
- [X] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [X] This pull-request is submitted against the `master` branch. 
  - [X] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [X] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

[Git Issue 1290](https://github.com/Azure/azure-iot-sdk-c/issues/1290)

# Description of the problem

These changes support the latest refresh of the reference docs located at [https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/](https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/)

* Added comments for doc coverage of [IoTHubMessage_SetProperty()](https://docs.microsoft.com/azure/iot-hub/iot-c-sdk-ref/iothub-message-h/iothubmessage-setproperty) to inform customers of the valid characters for the property key and value.
* Remove hardcoded english language in HTML links. These causes issues with docs.microsoft.com build and validation checking. By using non-locale specific links we retain the option to localize reference content/

# Description of the solution
 

Just added a bold note to help customers understand accepted values.

### CODE REVIEW

File: iothub_client/inc/iothub_message.h [hunk 1]
```diff
@@ -191,9 +191,12 @@ MOCKABLE_FUNCTION(, MAP_HANDLE, IoTHubMessage_Properties, IOTHUB_MESSAGE_HANDLE,
...
 * @param   iotHubMessageHandle Handle to the message.
 *
-* @param   key name of the property to set.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
+* @param   key name of the property to set. Note that when sending messages via the HTTP transport, this value must not contain spaces.
 *
-* @param   value of the property to set.
+* @param   value of the property to set. Note that when sending messages via the HTTP transport, this value must not contain spaces.
+*
+*            @b NOTE: Property names and values must not contain spaces and can only contain ASCII alphanumeric characters, 
+*            plus {'!', '#', '$', '%, '&', ''', '*', '+', '-', '.', '^', '_', '`', '|', '~'}.
```
Comment (ericwol-msft) [inline iothub_client/inc/iothub_message.h:199]: I think mollie is trying to validate this list.

**Review by ericwol-msft — APPROVED** [review]

File: iothub_client/inc/iothub_message.h (hunk 1 above)
Comment (momuno) [reply iothub_client/inc/iothub_message.h:199]: Working on this.  Please wait to merge this, as we may need to change what is written here.  Thanks.

**Comment by wesmc7777** [comment]
@momuno see if this looks good to you.   

The reference docs would look as follows once I push them live -> [IoTHubMessage_SetProperty()](https://review.docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-message-h/iothubmessage-setproperty?branch=pr-en-us-43)

File: iothub_client/inc/iothub_message.h [hunk 2]
```diff
@@ -191,9 +191,14 @@ MOCKABLE_FUNCTION(, MAP_HANDLE, IoTHubMessage_Properties, IOTHUB_MESSAGE_HANDLE,
...
+* @param   key name of the property to set. Note that when sending messages via the HTTP transport, this value must not contain spaces.
 *
-* @param   value of the property to set.
+* @param   value of the property to set. 
+*
+*            @b NOTE: Generally, property key names and values can only contain ASCII alphanumeric characters, 
+*            plus {'!', '#', '$', '%, '&', ''', '*', '+', '-', '.', '^', '_', '`', '|', '~'}. However, the accepted  
+*            character sets are dependent on other factors such as the protocol being used. For more information
+*            on the character sets accepted by Azure IoT Hub, see <a href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct">Create and read IoT Hub messages</a>.
 *
```
Comment (momuno) [inline iothub_client/inc/iothub_message.h:202]: Each protocol is different enough that I would just point them to the doc.  So something like:

NOTE: The accepted character sets are dependent on different factors, such as the protocol being used. For more information on the character sets accepted ... (the rest you have)

**Review by momuno — APPROVED** [review]

File: iothub_client/inc/iothub_message.h (hunk 2 above)
Comment (wesmc7777) [reply iothub_client/inc/iothub_message.h:201]: Ok, I trimmed it down as you recommended

Comment (momuno) [reply iothub_client/inc/iothub_message.h:201]: Thanks, looks good

**Pull request merged**